capa/engine.py

```diff
@@ -44,9 +44,9 @@ def __init__(self, description=None):
 
     def __str__(self):
         if self.description:
-            return "%s(%s = %s)" % (self.name.lower(), ",".join(map(str, self.get_children())), self.description)
+            return f"{self.name.lower()}({','.join(map(str, self.get_children()))} = {self.description})"
         else:
-            return "%s(%s)" % (self.name.lower(), ",".join(map(str, self.get_children())))
+            return f"{self.name.lower()}({','.join(map(str, self.get_children()))})"
```

Comment:
neither the old nor the new are very readable imho.

i think we should avoid putting much logic within the `{...}` blocks because its hard for humans to keep track of the context: its a string, no an interpolation, no function call, etc.

if the substitutions are complex, lets do something like:

```py
name = self.name.lower()
children = ','.join(map(str, self.get_children()))
return f"{name}({children} = {self.description})"
```

Reply:
im not sure how to quantify "complex", but i wonder about "if there's more than one manipulation or transformation, its complex"